api/proxy/store/eigenda_manager.go

```diff
@@ -91,18 +92,115 @@ func (m *EigenDAManager) Get(ctx context.Context,
 	versionedCert certs.VersionedCert,
 	opts common.GETOpts,
 ) ([]byte, error) {
-	if versionedCert.Version == certs.V0VersionByte && m.eigenda == nil {
-		return nil, errors.New("expected EigenDA V1 backend for DA commitment type with CertV0")
+	switch versionedCert.Version {
+	case certs.V0VersionByte:
+		if m.eigenda == nil {
+			return nil, errors.New("received CertV0 but EigenDA V1 client is not initialized")
+		}
+		return m.getEigenDAV1(ctx, versionedCert, opts)
+	case certs.V1VersionByte, certs.V2VersionByte:
+		if m.eigendaV2 == nil {
+			return nil, errors.New("received EigenDAV2 cert but EigenDA V2 client is not initialized")
+		}
+		return m.getEigenDAV2(ctx, versionedCert, opts)
+	default:
+		return nil, fmt.Errorf("cert version unknown: %b", versionedCert.Version)
+	}
+}
+
+// getEigenDAV1 will attempt to retrieve a blob for the given versionedCert
+// from cache, EigenDA V1, and fallback storage.
+// TODO: we should also add the v1 RetrievalClient to retrieve from the validators directly
+// in case the v1 disperser is down, the same way we do for v2.
+func (m *EigenDAManager) getEigenDAV1(
+	ctx context.Context,
+	versionedCert certs.VersionedCert,
+	opts common.GETOpts,
+) ([]byte, error) {
+	if opts.ReturnEncodedPayload {
+		return nil, fmt.Errorf("returning encoded payload is not supported for V0 certificates (EigenDA V1)")
 	}
-	if versionedCert.Version == certs.V1VersionByte && m.eigendaV2 == nil {
-		return nil, errors.New("expected EigenDA V2 backend for DA commitment type with CertV1")
+
+	verifyFnForSecondary := func(ctx context.Context, cert []byte, payload []byte, l1InclusionBlockNumber uint64) error {
+		// We don't add the cert version because EigenDA V1 only supports [certs.V0VersionByte] Certs.
+		// We also don't use the l1InclusionBlockNumber because Recency check is only supported by EigenDA V2.
+		// TODO: we should decouple the Verify function into a VerifyCert and VerifyBlob function,
+		// and only verify the cert once here, before retrievals, and then only verify the blob commitment
+		// after retrievals.
+		return m.eigenda.Verify(ctx, cert, payload)
 	}
 
-	// The eigendav2 verify function verifies the cert, but not the kzg commitment.
-	// TODO: should we be verifying those when retrieving from secondary storages like S3...?
-	verifyMethod, err := m.getVerifyMethod(versionedCert.Version)
+	var readErrors []error
+	// 1 - read payload from cache if enabled
+	// Secondary storages (cache and fallback) store payloads instead of blobs.
+	// For simplicity, we bypass secondary storages when requesting encoded payloads,
+	// since those requests are only for secure integrations and run by provers/challengers.
```

Comment:
> 	// For simplicity, we bypass secondary storages when requesting encoded payloads,
>	// since those requests are only for secure integrations and run by provers/challengers.

This isn't relevant for v1, right? You early-out above if `ReturnEncodedPayload` is set

Reply:
[53c47f3](https://github.com/Layr-Labs/eigenda/pull/1914/commits/53c47f373a208dccce1fbd163f4b0729bc26e1f8)